src/rtk/features/delegationRequest/delegationRequestSlice.ts

```diff
@@ -1,86 +1,111 @@
 import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 import axios from 'axios';
 
-import type { DelegableApi } from '../delegableApi/delegableApiSlice';
-import type { DelegableOrg } from '../delegableOrg/delegableOrgSlice';
+import { getCookie } from '@/resources/Cookie/CookieMethods';
 
 export interface ApiDelegation {
   orgName: string;
   apiName: string;
 }
 
-export interface InitialState {
+export interface SliceState {
   loading: boolean;
   error: string;
   succesfulApiDelegations: ApiDelegation[];
   failedApiDelegations: ApiDelegation[];
+  batchPostSize: number;
+  batchPostCounter: number;
 }
 
 export interface DelegationRequest {
   apiIdentifier: string;
+  apiName: string;
   orgNr: string;
+  orgName: string;
 }
 
-const initialState: InitialState = {
+const initialState: SliceState = {
   loading: true,
   error: '',
-  succesfulApiDelegations: [
-    {
-      orgName: 'Org blabla',
-      apiName: 'Api blabla',
-    },
-  ],
-  failedApiDelegations: [
-    {
-      orgName: 'Org blabla',
-      apiName: 'Api blabla',
-    },
-  ],
+  succesfulApiDelegations: [],
+  failedApiDelegations: [],
+  batchPostSize: 0,
+  batchPostCounter: 0,
 };
 
-export const delegateApi = async (
-  state: InitialState,
-  apiList: DelegableApi[],
-  orgList: DelegableOrg[],
-) => {
-  for (let i = 0; i < apiList.length; i++) {
-    for (let j = 0; j < orgList.length; j++) {
-      const delegationRequest = {
-        apiIdentifier: apiList[i].id,
-        orgNr: orgList[j].orgNr,
-      };
-
-      const apiDelegation = {
-        apiName: apiList[i].apiName,
-        orgName: orgList[j].orgName,
-      };
-      try {
-        await postApiDelegation(delegationRequest);
-        state.succesfulApiDelegations.push(apiDelegation);
-      } catch {
-        state.failedApiDelegations.push(apiDelegation);
-      }
-    }
-  }
-};
-
-const postApiDelegation = createAsyncThunk(
-  'delegationRequestSlice/delegateApi',
-  async ({ apiIdentifier, orgNr }: DelegationRequest) => {
+export const postApiDelegation = createAsyncThunk(
+  'delegationRequestSlice/postApiDelegation',
+  async (delegationInfo: DelegationRequest) => {
+    const { apiIdentifier, apiName, orgNr, orgName }: DelegationRequest = delegationInfo;
+    const delegation: ApiDelegation = {
+      apiName,
+      orgName,
+    };
+    const altinnPartyId =
+      getCookie('AltinnPartyId') === null ? getCookie('AltinnPartyId') : '50067798';
```

Comment:
This is opposite of what it should be. Don't default to 50067798 when not cookie is null, throw Error instead.

Reply:
Isn't this how we've had it for a while because of the whole httpOnly thing? 😅

But now that we can just turn that off in the browser loacally, I guess we can remove the default-values we've litterede around in our code - I can do that in this PR 🙂

Reply:
I don't know. I fixed the issue in my PR in issue 32 so no worries about this comments